### ISSUE TITLE
Links

### ISSUE THREAD

**Issue by XieEDeHeiShou** [vsch/idea-multimarkdown#741]
How to config <kbd>ctrl</kbd>+<kbd>left click</kbd> on a relative path as `open file` like default markdown support plugin instead of `show usage` in editor plane?

I see these a similar issue which related #317 , but I'm not going to export markdown as HTML.

IntelliJ IDEA 2019.1.1 (Ultimate Edition) with Markdown Navigator v2.8.4.

**Comment by vsch** [comment]
@XieEDeHeiShou, can you please give me specific example where this does not work.

By default the IDE ctrl+left click (cmd+left click on OS X) performs show usages/go to definition.

I tried it and it does work correctly, however it is possible that there is an issue with your specific configuration or relative path.

If you perform `Go to declaration` on the same link does it open the file?

You can also try an EAP release to see if it is related to another fixed issue:

[Installing EAP Plugin Version](https://github.com/vsch/idea-multimarkdown/wiki/EAP-Update-Channel)

**Comment by XieEDeHeiShou** [comment]
Markdown support:
![markdown-support](https://user-images.githubusercontent.com/16363180/56401169-37a3c780-628a-11e9-9040-2486946b07b0.gif)

Markdown navigator stable:
![markdown-navigator](https://user-images.githubusercontent.com/16363180/56401177-44c0b680-628a-11e9-9c59-eac90288cecb.gif)

Markdown navigator eap: (the same as stable)
![markdown-navigator-eap](https://user-images.githubusercontent.com/16363180/56401195-4db18800-628a-11e9-804f-2a49449f178e.gif)

Markdown navigator eap with go to actions:
![markdown-navigator-eap-go-to-action](https://user-images.githubusercontent.com/16363180/56401410-3c1cb000-628b-11e9-81e9-373deb84980a.gif)

**Comment by vsch** [comment]
@XieEDeHeiShou, from the links it looks like they are referring to files outside of the current project.

If this is true then it would explain why the links do not work to open files. The plugin only resolves page relative links if they are part of the project. 

I will try to see if there is a workaround for this in the current implementation.

**Comment by XieEDeHeiShou** [comment]
Please mark this issue with a bug label.

**Comment by fuzzyweapon** [comment]
@vsch I seem to **actually** have a bug (mind you I am not using premium version so I'm not sure if that matters here**

The following link styles I've tried do not work using @XieEDeHeiShou's test cases (note that all documents in these tests exist in the root of the project and all anchors exist):
- `[Display Name](FAQ)`
- `[Display Name](FAQ#my-anchor)`
- `[Display Name](./FAQ)`
- `[Display Name](./FAQ#my-anchor)`

**Comment by fuzzyweapon** [comment]
To follow up, this is kind of a global thing for me it seems.  These also fail.
- `[[FAQ]]`
![image](https://user-images.githubusercontent.com/1253399/57898077-599e5480-780c-11e9-8566-dfce2cbd3d29.png)
If I try to navigate to it through usages, it does nothing.
If I try to use the intention action to fix it by making the link explicit, nothing happens.

- `[FAQ]`
![image](https://user-images.githubusercontent.com/1253399/57898221-f19c3e00-780c-11e9-8371-e22fcb55b42e.png)
If I try to navigate to it through usages, it does nothing.
There is no intention action available outside of (`Convert line/selected to comment` and `Change link to test`)
This type of link is never rendered by GH ever.  I realize this plugin doesn't just handle GH, but I kinda figured (perhaps wrongly) that it was already configured toward GFM and trying to render it as closely to GH as possible in the preview?

In summary, I feel like something is deeply wrong with my install given not everyone is having these problems?  Have you ever seen such things?
I only have Markdown-Navigator enabled and no other navigation related or Markdown related plugins enabled.

**Comment by vsch** [comment]
@fuzzyweapon, there are several parts to your issue.

1. `[[FAQ]]` is a wiki link and on GitHub these are only valid in wiki pages. The plugin requires that a wiki directory be configured in a specific way so that all markdown files under this directory are treated as a wiki for the project: [Adding-GitHub-Wiki-to-IntelliJ-Project]

2. `[FAQ]` is a reference link to a reference defined somewhere in the file. You must have the following, preferably with a blank line preceding it:

       [FAQ]: FAQ-file-url

3. Files without extensions need extra logic for processing them since they are not associated with any file type, other than plain text. The plugin uses file type index, provided by the IDE, to get a cached list of files in the project. This has a lot of caveats, one of which requires that the files are in some form part of the project: source, resource, test source, test resource, otherwise the IDE will not include them in the file type index.

   There is also a bug which in some cases will not get a file without extension listed as part of "Plain Text" files, which is how the IDE treats files without extensions by default. Additionally, default file type was Markdown which did not include plain text file, so no files without extension would not be listed in completions. Both are fixed for next release.

   I will update this issue when the EAP version with the fix is available.

In general the plugin follows GitHub resolution rules including all caveats of Wiki link resolution in GitHub Wiki repositories. So if you have a discrepancy it is either a bug or a misconfiguration of plugin settings.

[Adding-GitHub-Wiki-to-IntelliJ-Project]: https://github.com/vsch/idea-multimarkdown/wiki/Adding-GitHub-Wiki-to-IntelliJ-Project

**Comment by vsch** [comment]
Plain text files now showing up in completions or showing as unresolved, EAP update with a fix for the issue is available.

[Installing EAP Plugin Version](https://github.com/vsch/idea-multimarkdown/wiki/EAP-Update-Channel)

**Comment by vsch** [comment]
When you try the latest release with version 2019.3 of the IDE, please let me know if the issue is resolved.
